openedx/features/colaraz_features/middlewares.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+from django.conf import settings
+from django.contrib.auth import logout
+from django.http import HttpResponseRedirect, Http404
+from django.urls import reverse
+from six.moves.urllib.parse import urlencode
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from student.models import UserAttribute
+import third_party_auth
+from third_party_auth import pipeline
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ColarazAuthenticationMiddleware(object):
+    """
+    Special middleware for Colaraz which does following tasks:
+    1) Verify User's accessibility over his requested site
+    2) Login's User if he is logged-in on Colaraz's website
+    """
+
+    def process_request(self, request):
+        """
+        Django middleware hook for processing request
+        """
+        if getattr(settings, 'COLARAZ_ENABLE_AUTH_MIDDLEWARE', False):
+            user = request.user
+            
+            # Blocking all the paths which need to be shown to logged in users only
+            blocked_sub_paths = getattr(settings, 'COLARAZ_BLOCKED_SUB_PATHS', [])
+            blocked_full_paths = getattr(settings, 'COLARAZ_BLOCKED_FULL_PATHS', [])
+            is_blocked = [blocked_path in request.path for blocked_path in blocked_sub_paths]
+            is_blocked += [blocked_path == request.path for blocked_path in blocked_full_paths]
+
+            if not user.is_authenticated and any(is_blocked):
+                LOGGER.info('Requested path "{}" is blocked for anonymous users'.format(request.path))
+                return self._redirect_to_login(request)
```

Comment:
Here do we need to handle the case when user is not authenticated but `is_blocked` is all `False`.?

Reply:
Nope.